storm-server/src/main/java/org/apache/storm/localizer/AsyncLocalizer.java

```diff
@@ -365,6 +384,7 @@ void addReferencesToBlobs(PortAndAssignment pna, BlobChangingCallback cb)
         if (!localResourceList.isEmpty()) {
             getBlobs(localResourceList, pna, cb);
         }
+        pna.complete();
```

Comment:
just one more nit: put this outside of this method since the method `addReferencesToBlobs` doesn't indicate `complete`